base/strings/util.jl

```diff
@@ -275,9 +275,9 @@ split(str::T, splitter::Char;
     _split(str, equalto(splitter), limit, keep, T <: SubString ? T[] : SubString{T}[])
 
 function _split(str::AbstractString, splitter, limit::Integer, keep_empty::Bool, strs::Array)
-    i = start(str)
+    @assert (i = firstindex(str)) == 1
```

Comment:
Do we really need to assert this? Why do it here and not elsewhere? I think implementations which get this wrong will realize it quite soon even without assertions given that may things will fail.

Reply:
I'm asserting it because the subsequent code relies on it.  Basically, I think of this as a comment with some added value.  Assertions are cheap so we shouldn't hesitate to add them, no?

Reply:
`(i = firstindex(str))` though goes against the recently added documention that use of `@assert` might not work in the future (#25610)

Reply:
Okay, will fix.

Reply:
Not really since the argument here is to delete it which is exactly what the compiler would do.

Reply:
@StefanKarpinski, the point is that you can't completely delete this code because it contains an assignment.  Hence this is not correct usage of `@assert`.

Reply:
Ah, yes. That's true. I missed that bit.